test/integration/concurrency_test.py

```diff
@@ -0,0 +1,49 @@
+import ssl
+import pytest
+import pyexasol
+import threading
+from pyexasol import ExaConcurrencyError
+
+
+class QueryThread(threading.Thread):
+
+    def __init__(self, connection, timeout):
+        self.connection = connection
+        self.seconds = timeout
+        self.exception = None
+        super().__init__()
+
+    def run(self):
+        try:
+            # run heavy query
+            self.connection.execute("SELECT * FROM users a, users b, users c, payments d")
```

Comment:
Could the query be simplified?
Or is there a specific reason for querying table `users` 3 times?

Reply:
took this from the original example, see comment above. 
I assume this shall make sure query does take long enough for the second attempt.
Maybe it can be done with a sleep, I'll try.

Reply:
Previously "sleep" UDFs did not work well with abort_query() call. Usually it caused an entire connection to be terminated.

Reply:
@littleK0i from what I have seen >= 7.1, sleep works fine.
With this information which query would you prefer and why?